routes/api/category-routes.js

```diff
@@ -0,0 +1,77 @@
+const router = require('express').Router();
+const { Category, Product } = require('../../models');
+
+// The `/api/categories` endpoint
+
+router.get('/', async (req, res) => {
+  try {
+    const categoryData = await Category.findAll();
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+  // be sure to include its associated Products
+});
+
+router.get('/:id', async (req, res) => {
+  const id = req.params.id
+  // find one category by its `id` value
+  // be sure to include its associated Products
+  try {
+    const categoryData = await Category.findByPk(id);
+    if (!categoryData) {
+      res.status(404).json({ message: 'No Category found with this id!' });
+      return;
+    }
+
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+router.post('/', async (req, res) => {
+  try {
+    const categoryData = await Category.create(req.body);
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(400).json(err);
```

Comment:
Consider responding with a 500 error here and additing some validation before line 35 to ensure the user has sent all of the required infromation, if not responding with a 400 error then e.g.
```
    const { category_name } = req.body
    
    if (!category_name) {
      return res.status(400).json({
        error: "Values undefined",
        message: "Please provide a category name."
      })
    }
```